### ISSUE TITLE
ctrl+f вставляет лишнее

### ISSUE THREAD

**Issue by peter-volkov** [michaellukashov/Far-NetBox#162]
При работе с netbox 2.1.41 (build 365) в панели по нажатию ctrl+f он вставляет ssh://<login>:@<server>:22/<file_path> вместо просто <file_path> как делается в WinSCP или при работе с локальной файловой системой.

Возможно, кто-то работает именно с таким сценарием использования, но я лично привык пользоваться ctrl+f чтобы заполнять аргументы консольных команд полными путями файлов в панели для запуска. Мне приходится руками удалять ssh схему (ssh://<login>:@<server>:22/) для каждого файла, чтобы запускать консольные команды из панели. Мягко говоря, неудобно.

Если ли возможность вставлять полные пути файлов из панели в консоль без ssh схемы?
Если нет, можно ли сделать такую опцию?

**Comment by VictorVG** [comment]
Да, где-то поломался механизм копирования (README.RU.MD::4) URL:
>4. Фичи
>
>Комбинация клавиш в панели:
>Ctrl+Alt+Ins: Копирование текущего URL в буфер обмена (вместе с паролем).
>Shift+Alt+Ins: Копирование текущего URL в буфер обмена (без пароля).

и ShiftAltIns срабатывает аналогично CtrlAltIns - 

CtrlAltIns: ftp://anonymous:@ftp5.gwdg.de:21/pub/tdf/libreoffice/testing/5.0.1/win/x86_64/LibreOffice_5.0.1.2_Win_x64.msi
ShiftAltIns ftp://anonymous:@ftp5.gwdg.de:21/pub/tdf/libreoffice/testing/5.0.1/win/x86_64/LibreOffice_5.0.1.2_Win_x64.msi вместо ожидаемого ftp://ftp5.gwdg.de/pub/tdf/libreoffice/testing/5.0.1/win/x86_64/LibreOffice_5.0.1.2_Win_x64.msi

и это надо чинить и ваша проблема с ним связана. Спасибо что заметили.

**Comment by peter-volkov** [comment]
Спасибо за ответ. Мне кажется, в #163 речь несколько об ином, а описанное в этом таске поведение присутствует в NetBox еще до того, как NetBox вошел в базовую поставку Far Manager,
#163 жалоба на наличие логина и порта в URL, я же прошу сделать по Ctrl+F вставку полного пути до файла в локальной системе, вообще без задания схемы ssh:// или ftp://
то есть вместо ssh://root/script.sh и, тем более, ssh://login@servername:port/root/script.sh хочется вставлять в консоль просто /root/script.sh

**Comment by peter-volkov** [comment]
Для ftp это может быть оправданно, но при использовании ssh, я активно пользуюсь удалённой консолью.
В ней, кстати, еще ряд проблем - не работает интерактивная консоль (например, когда sudo запрашивает пароль) и в ответе какие-то целые числа посторонние добавляются.
Я бы мог поизучать и прислать пул-реквесты, но теряюсь сходу в коде - если бы кто-нибудь мини-экскурсию провёл, я бы мог покрутить.

**Comment by VictorVG** [comment]
С интерактивной консолью понятно что происходит - новый процесс, родитель в ожидании и видимо не возврат управления. С мусором стоит логи посмотреть иначе трудно представить место откуда он может браться. 

С линками - тут по идее чинить консерваторию надо, а вот когда именно она сломалась сейчас установить сложно - у меня в коллекции далеко не все сборки Far3 есть. Я обычно когда его исходники обновляются старую сборку в архив на NFS ссылаю и они там лежат для поиска разных бяк, а NetBox там по мере возможности есть, так что по всем ревизиям не пробежаться...

**Comment by peter-volkov** [comment]
Мне вот непонятно. Новый процесс где? В локальной системе рабочей станции при повисании из-за интерактивной консоли новых процессов не появляется - я проверил. 
У standalone gui версии WinSCP, к слову, та же проблема.
Подозреваю, на уровне лока между потоками дело.

Про мусор. Пример:
$ date
Thu Aug 27 18:58:25 UTC 2015
0
$ cat /etc/passwd | wc -l
25
0

То есть к концу ответа 0 дописывается. Бывают, вроде, и другие цифры. 
В отладочный лог NetBox заглянул - там ничего интересного.
Всё в духе
. 2015-08-27 21:46:42.826 Received 41 bytes (0)
. 2015-08-27 21:46:42.827 Read 4 bytes (37 pending)
. 2015-08-27 21:46:42.828 Read 37 bytes (0 pending)


Второй абзац не понял. Ни про линки, ни про консерваторию, ни про смысл пробегания по всем сборкам Far3.

**Comment by VictorVG** [comment]
Второй абзац прост - NB умеет копировать URL в двух вариантах - с паролями и без, а это и сломалось - посмотрите #163 .

Со сборками - мои скрипты собирают Far из SVN, а там NetBox нет и он попадает в сборки если совпало по времени появление новой версии Far и NetBox и по таким архивам (я использую 7-Zip) можно попробовать найти примерное время возникновения поломки, но на сегодня у меня их "всего" 495 штук, так что объём возни думаю вы представляете.

С локом - тут похоже как раз ошибка WinSCP вылезла ибо:

>Based on WinSCP as FAR Plugin: SFTP/FTP/SCP client for FAR version 1.6.2 Copyright (c) 2000-2009 Martin Prikryl
>Based on WinSCP version 5.7.5 Copyright (c) 2000-2015 Martin Prikryl
>SSH and SCP code based on PuTTY 0.65 Copyright (c) 1997-2015 Simon Tatham
>FTP code based on FileZilla 2.2.32 Copyright (c) 2001-2007 Tim Kosse

значит чинить нужно скорее всего этот модуль - вот вам и "консерватория" .:)
